observations/obs_converters/MPD/text_to_obs.f90

```diff
@@ -0,0 +1,206 @@
+
+obsloop: do    ! no end limit - have the loop break when input ends
+
+   ! read in a line from the text file.   What you need to create an obs:
+   !  location: lat, lon, and height in pressure or meters
+   !  time: when the observation was taken
+   !  type: from the DART list of obs types
+   !  error: very important - the instrument error plus representativeness error
+   !        (see html file for more info)
+
+   ! assume here a line is a type, location, time, value, obs error
+
+   ! read in entire text line into a buffer
+   read(iunit, "(A)", iostat=rcio) input_line
+   if (rcio /= 0) then
+      if (debug) print *, 'got bad read code from input file, rcio = ', rcio
+      exit obsloop
+   endif
+
+   ! pull off the first value as an integer, to decode the type
+   read(input_line, *, iostat=rcio) otype
+   if (rcio /= 0) then
+      if (debug) print *, 'got bad read code trying to get obs type, rcio = ', rcio
+      exit obsloop
+   endif
+
+   if (debug) print *, 'next observation type = ', otype
+
+   ! for this example, assume there is an obs type, where otype=1 is
+   ! abs humidity, the input text file has these as their own hardcoded convention.
+
+   if (otype == 1) then !abs_humidity
+      read(input_line, *, iostat=rcio) otype, lat, lon, vert, &
+                                 year, month, day, hour, minute, second, &
+                                 abs_humid, terr
+      if (rcio /= 0) then
+         if (debug) print *, 'got bad read code getting rest of abs_humidity obs, rcio = ', rcio
+         exit obsloop
+      endif
+
+   else !WHAT?
+
+      ! no method defined to convert this type
+      write(*,*) "NOOOOO, DON'T DO THIS: no method defined to convert this type = ", otype
+
+   endif
+
+   if (debug) print *, 'next observation located at lat, lon = ', lat, lon
+
+   ! check the lat/lon values to see if they are ok
+   if ( lat >  90.0_r8 .or. lat <  -90.0_r8 ) cycle obsloop
+   if ( lon <   0.0_r8 .or. lon >  360.0_r8 ) cycle obsloop
+
+   ! put date into a dart time format
+   time_obs = set_date(year, month, day, hour, minute, second)
+
+   if (debug) call print_date(time_obs, 'next obs time is')
+
+   ! extract time of observation into gregorian day, sec.
+   call get_time(time_obs, osec, oday)
+
+   ! this example assumes there is an obs type, where otype=1 is
+   ! abs_humidity measured in height
+   if (otype == 1) then !abs_humidity
+
+      ! height is in meters
+      ! make an obs derived type, and then add it to the sequence
+      call create_3d_obs(lat, lon, vert, VERTISHEIGHT, abs_humid, &
+                         MPD_ABSOLUTE_HUMIDITY, terr, oday, osec, qc, obs)
+      call add_obs_to_seq(obs_seq, obs, time_obs, prev_obs, prev_time, first_obs)
+      if (debug) print *, 'added abs humidity obs to output seq'
+
+   else !WHAT???
+
+      ! no method defined to convert this type
+      write(*,*) 'no method defined to convert this type = ', otype
```

Comment:
So is this an error or what? Should the program stop?

Reply:
this should be a warning only, so I modified the warning message